config/default/nools-extras.js

```diff
@@ -0,0 +1,314 @@
+      result = report;
+    }
+  });
+  return result;
+};
+
+const getLMPDateFromPregnancy = function (report) {
+  return isPregnancyForm(report) &&
+   getDateMS(getField(report, 'lmp_date_8601'));
+};
+
+const getLMPDateFromPregnancyFollowUp = function (report) {
+  return isPregnancyFollowUpForm(report) &&
+   getDateMS(getField(report, 'lmp_date_8601'));
+};
+
+
+
+function getSubsequentPregnancies(contact, refReport) {
+  return contact.reports.filter(function (report) {
+    return isPregnancyForm(report) && report.reported_date > refReport.reported_date;
+  });
+}
+
+function getSubsequentPregnancyFollowUps(contact, report) {
+  const subsequentVisits = contact.reports.filter(function (visit) {
+    let lmpDate = getLMPDateFromPregnancy(report);
+    if (!lmpDate) { //LMP Date is not available use reported date
+      lmpDate = report.reported_date;
+    }
+
+    return isPregnancyFollowUpForm(visit) &&
+      visit.reported_date > report.reported_date &&
+      visit.reported_date < addDays(lmpDate, MAX_DAYS_IN_PREGNANCY);
+  });
+  return subsequentVisits;
+}
+
+function getSubsequentDeliveries(contact, refReport, withinLastXDays) {
+  return contact.reports.filter(function (deliveryReport) {
+    return (deliveryReport.form === 'delivery') &&
+      deliveryReport.reported_date > refReport.reported_date &&
+      (!withinLastXDays || refReport.reported_date >= (today - withinLastXDays * MS_IN_DAY));
+  });
+}
+
+function getMostRecentLMPDateForPregnancy(contact, report) {
+  let mostRecentLMP = getLMPDateFromPregnancy(report);
+  let mostRecentReportDate = report.reported_date;
+  getSubsequentPregnancyFollowUps(contact, report).forEach(function (v) {
+    const lmpFromPregnancyFollowUp = getLMPDateFromPregnancyFollowUp(v);
+    if (v.reported_date > mostRecentReportDate && lmpFromPregnancyFollowUp !== '' && lmpFromPregnancyFollowUp !== mostRecentLMP) {
+      mostRecentReportDate = v.reported_date;
+      mostRecentLMP = lmpFromPregnancyFollowUp;
+    }
+  });
+  return mostRecentLMP;
+}
+
+
+function isPregnancyTerminatedByAbortion(contact, report) {
+  const followUps = getSubsequentPregnancyFollowUps(contact, report);
+  const latestFollowup = getNewestReport(followUps, antenatalForms);
+  return latestFollowup && getField(latestFollowup, 'pregnancy_summary.visit_option') === 'abortion';
+}
+
+function isPregnancyTerminatedByMiscarriage(contact, report) {
+  const followUps = getSubsequentPregnancyFollowUps(contact, report);
+  const latestFollowup = getNewestReport(followUps, antenatalForms);
+  return latestFollowup && getField(latestFollowup, 'pregnancy_summary.visit_option') === 'miscarriage';
+}
+
+function isActivePregnancy(contact, report) {
+  if (!isPregnancyForm(report)) return false;
+  const lmpDate = getMostRecentLMPDateForPregnancy(contact, report);
+  return lmpDate > today - MAX_DAYS_IN_PREGNANCY * MS_IN_DAY && //Pregnancy registration in the past 9 months
```

Comment:
In general, we try to have comments that describe "why" things are, but not "what" things are. If you need comments for "what some code does" it probably can be broken up or clarified. 

One way to make this code "self documenting" is to assign each of these into variables.

```
const isPregnancyRegisteredWithin9Months = ...;
const isPregnancyTerminatedByDeliveryIn6Weeks = ...;
const isPregnancyTerminatedByAnotherPregnancyReport = ...;
...

return isPregnancyRegisteredWithin9Months && isPregnancyTerminatedByDeliveryIn6Weeks && isPregnancyTerminatedByAnotherPregnancyReport ...;
```

Reply:
Thanks for the suggestion, I have updated the code using variables. However, there is one thing that I am concerned about:
Previously, if the first condition was falsy, then the function would return false without having to check the second and subsequent conditions.
Now, as we are using variables, it has to check everything beforehand. Is it worth avoiding these unnecessary checks?

Reply:
Good point. Maybe you had it right the first time. I'll let you decide. Sorry to randomize you.